vm/src/builtins/type.rs

```diff
@@ -594,6 +596,19 @@ impl PyType {
         let typ = Self::new_verbose_ref(name.as_str(), base, bases, attributes, slots, metatype)
             .map_err(|e| vm.new_type_error(e))?;
 
+        let __classcell__ = identifier!(vm, __classcell__);
+
+        // It is correct to get it from tp_dict, not attributes.
+        let cell = typ
+            .attributes
+            .write()
+            .get(__classcell__)
+            .map(|cell| PyCellRef::try_from_object(vm, cell.clone()));
+
+        if let Some(Ok(cell)) = cell {
+            cell.set(Some(typ.clone().to_pyobject(vm)));
+        }
+
```

Comment:
```suggestion
        typ.attributes
            .write()
            .get(identifier!(vm, __classcell__))
            .map(|cell| {
                if let Some(cell) = PyCellRef::try_from_object(vm, cell.clone()) {
                    cell.set(Some(typ.clone().to_pyobject(vm)));
                    Ok(())
                } else {
                    Err(vm.new_type_error(format!(
                        "__classcell__ must be a nonlocal cell, not {}",
                        cell.class()
                    )))
                }
            })
            .transpose()?;

```
It seems that we miss to raise type error when the given `__classcell__` is actually not a cell type

Reply:
I think previous version is better. `cell.set` has side effect and the result of `map` is not used. This is more like control flow than data stream. Using `if-let` will show the flow better and will not hide side-effect of `cell.set`.

Reply:
Oh I agree. Then we only need to add type error handling